mutex_art.hpp

```diff
@@ -21,18 +21,19 @@ namespace unodb {
 /// global lock and hold it for the duration of the operation.
 ///
 /// \sa unodb::olc_art for a highly concurrent thread-safe ART implementation.
-template <typename Key>
+template <typename Key, typename Value>
 class mutex_db final {
  public:
   using key_type = Key;
+  using value_type = Value;
   using value_view = unodb::value_view;
 
   /// If the search key was found, that is, the first pair member has a value,
   /// then the second member is a locked tree mutex which must be released ASAP
   /// after reading the first pair member. Otherwise, the second member is
   /// undefined.
-  using get_result =
-      std::pair<typename db<Key>::get_result, std::unique_lock<std::mutex>>;
+  using get_result = std::pair<typename db<Key, Value>::get_result,
```

Comment:
get_result here is defined with Value rather than value_view (which is used in art.hpp, olc_art.hpp).